modules/integration/tests-iot-web-ui/src/test/java/org/wso2/carbon/iot/integration/web/ui/test/extension/IOTServerExtension.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.iot.integration.web.ui.test.extension;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.ContextXpathConstants;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.exceptions.AutomationFrameworkException;
+import org.wso2.carbon.automation.engine.extensions.ExecutionListenerExtension;
+import org.wso2.carbon.automation.extensions.ExtensionConstants;
+import org.wso2.carbon.automation.extensions.servers.carbonserver.CarbonServerExtension;
+
+import javax.xml.xpath.XPathExpressionException;
+
+/**
+ * Test Automation server extension to start the IOT core.
+ * This will set the carbon_home to {carbonHome}/core and port offset : 0
+ */
+public class IOTServerExtension extends ExecutionListenerExtension {
+
+    private CustomTestServerManager serverManager;
+    private static final Log log = LogFactory.getLog(CarbonServerExtension.class);
+    private String executionEnvironment;
+    private AutomationContext automationContext;
+
+
+    @Override
+    public void initiate() throws AutomationFrameworkException {
+        try {
+            automationContext = new AutomationContext("IOT", TestUserMode.SUPER_TENANT_USER);
+            if(getParameters().get(ExtensionConstants.SERVER_STARTUP_PORT_OFFSET_COMMAND) == null) {
+                getParameters().put(ExtensionConstants.SERVER_STARTUP_PORT_OFFSET_COMMAND, "0");
```

Comment:
is port offset 0 needed? Also need to define the magic number

Reply:
This is fixed in https://github.com/wso2/product-iots/pull/1454/commits/caba24a1e13612fd30c8e2befa5c5d713a64dee8